source/base/timer.cc

```diff
@@ -37,6 +38,83 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace internal
+{
+  namespace Timer
+  {
+    namespace
+    {
+      /**
+       * Type trait for checking whether or not something is a duration.
+       */
+      template <typename T>
+      struct is_duration : std::false_type {};
+
+      /**
+       * Specialization for durations.
+       */
+      template <typename Rep, typename Period>
+      struct is_duration<std::chrono::duration<Rep, Period>> : std::true_type {};
```

Comment:
All duration types inherit from some instantiation of `std::ratio<M,N>` (which gives the `period`; the underlying integral type is `Rep`). I am not sure about the name of the private value member accessed by `count()`; did you mean something else by '`duration` member'?